source/common/network/happy_eyeballs_connection_impl.h

```diff
@@ -98,6 +98,13 @@ class HappyEyeballsConnectionImpl : public ClientConnection,
   void hashKey(std::vector<uint8_t>& hash_key) const override;
   void dumpState(std::ostream& os, int indent_level) const override;
 
+  // Returns a new vector containing the contents of |address_list| sorted
+  // with address families interleaved, as per Section 4 of RFC 8305, Happy
+  // Eyeballs v2. It is assumed that The list must already sorted as per
+  // Section 6 of RFC6724, which happens in ares_getaddrinfo().
```

Comment:
does the Apple implementation also do this? If not, then we would be breaking this requirement for iOS

Reply:
Yes, it does, according to what I read online.

Reply:
maybe comment as such.
is there any way to CHECK on this, in case there are other DNS impls that might not realize the constraint?